pygmt/tests/test_figure.py

```diff
@@ -377,13 +377,50 @@ class TestSetDisplay:
 
     def test_set_display(self):
         """
-        Test if pygmt.set_display updates the SHOW_CONFIG variable correctly.
+        Test if pygmt.set_display updates the SHOW_CONFIG variable correctly and
+        Figure.show opens the preview image in the correct way.
         """
-        default_method = SHOW_CONFIG["method"]  # Current default method
-
-        for method in ("notebook", "external", "none"):
-            set_display(method=method)
-            assert SHOW_CONFIG["method"] == method
+        default_method = SHOW_CONFIG["method"]  # Store the current default method.
+
+        fig = Figure()
+        fig.basemap(region=[0, 3, 6, 9], projection="X1c", frame=True)
+
+        # Test the "notebook" display method.
+        set_display(method="notebook")
+        assert SHOW_CONFIG["method"] == "notebook"
+        if _HAS_IPYTHON:
+            with (
+                patch("IPython.display.display") as mock_display,
+                patch("pygmt.figure.launch_external_viewer") as mock_viewer,
+            ):
+                fig.show()
+                assert mock_viewer.call_count == 0
```

Comment:
Instead of checking `mock_display.call_count == 1`, we can also use `mock_display.assert_called_once`, but I personally feel `call_count` is more flexible and readable.